x/pool-incentives/keeper/grpc_query_test.go

```diff
@@ -273,6 +273,164 @@ func (s *KeeperTestSuite) TestIncentivizedPools() {
+
+	for _, tc := range []struct {
+		desc                 string
+		poolCreated          bool
+		internalGaugeWeights []sdk.Int
+		externalGauges       []externalGauge
+		clPoolWithGauge      bool
+		clGaugeWeight        sdk.Int
+
+		expectedNumExternalGauges int
+		expectedGaugeIDs          []uint64
+	}{
+		{
+			desc:                 "No pool exist",
+			poolCreated:          false,
+			internalGaugeWeights: []sdk.Int{},
+
+			expectedNumExternalGauges: 0,
+		},
+		{
+			desc:                 "All gauges are internal (no external gauges)",
+			poolCreated:          true,
+			internalGaugeWeights: []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
+
+			expectedNumExternalGauges: 0,
+		},
+		{
+			desc:                 "Mixed internal and external gauges",
+			poolCreated:          true,
+			internalGaugeWeights: []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
+			externalGauges:       []externalGauge{{isPerpetual: true}, {isPerpetual: false}},
+
+			// Since there is no concentrated pool, there are only 3 internal gauges, one for each lockup duration
+			expectedGaugeIDs:          []uint64{4, 5},
+			expectedNumExternalGauges: 2,
+		},
+		{
+			desc:                 "More external gauges than internal gauges",
+			poolCreated:          true,
+			internalGaugeWeights: []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
+			externalGauges:       []externalGauge{{isPerpetual: true}, {isPerpetual: false}, {isPerpetual: true}, {isPerpetual: true}, {isPerpetual: false}},
+
+			// Since there is no concentrated pool, there are only 3 internal gauges, one for each lockup duration
+			expectedGaugeIDs:          []uint64{4, 5, 6, 7, 8},
+			expectedNumExternalGauges: 5,
+		},
+		{
+			desc:                 "Same number of external gauges as internal gauges",
+			poolCreated:          true,
+			internalGaugeWeights: []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
+			externalGauges:       []externalGauge{{isPerpetual: true}, {isPerpetual: false}, {isPerpetual: true}},
+
+			// Since there is no concentrated pool, there are only 3 internal gauges, one for each lockup duration
+			expectedGaugeIDs:          []uint64{4, 5, 6},
+			expectedNumExternalGauges: 3,
+		},
+		{
+			desc:                 "Internal gauge for concentrated pool exists",
+			poolCreated:          true,
+			clPoolWithGauge:      true,
+			clGaugeWeight:        sdk.NewInt(100),
+			internalGaugeWeights: []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
+			externalGauges:       []externalGauge{{isPerpetual: true}, {isPerpetual: false}, {isPerpetual: true}, {isPerpetual: true}, {isPerpetual: false}},
+
+			// Since there is no concentrated pool, there are 4 internal gauges, one for each lockup duration and one for the epoch duration
+			// (Note that in our test defaults, the epoch duration does not overlap with any lockup durations)
+			expectedGaugeIDs:          []uint64{5, 6, 7, 8, 9},
+			expectedNumExternalGauges: 5,
+		},
```

Comment:
```suggestion
		{
			desc:                 "Internal gauge for concentrated pool exists",
			poolCreated:          true,
			clPoolWithGauge:      true,
			clGaugeWeight:        sdk.NewInt(100),
			internalGaugeWeights: []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
			externalGauges:       []externalGauge{{isPerpetual: true}, {isPerpetual: false}, {isPerpetual: true}, {isPerpetual: true}, {isPerpetual: false}},

			// Since there is a concentrated pool, there are 4 internal gauges, one for each lockup duration and one for the epoch duration
			// (Note that in our test defaults, the epoch duration does not overlap with any lockup durations)
			expectedGaugeIDs:          []uint64{5, 6, 7, 8, 9},
			expectedNumExternalGauges: 5,
		},
```

Reply:
I'm committing this so that you don't have to wait for CI and can merge ASAP

Reply:
NVM, seeing you're still committing to this branch so going to leave as is

Reply:
Fixed